app/pendrive_detection/pendrive_detector.py

```diff
@@ -0,0 +1,125 @@
+import os
+import sys
+from typing import Optional
+import psutil
+
+if sys.platform == "linux":
+    import pyudev
+
+if sys.platform == "win32":
+    import wmi
+
+
+class PenDriveFinder:
+    """
+    A utility class to detect pen drives and locate private key files (*.pem) stored
+    on them on various operating systems(Windows, Linux, macOS).
+    """
+
+    def __init__(self):
+        self.wmi_client = None
+
+    @staticmethod
+    def check_os() -> str:
+        """
+        Determines the current operating system (Linux, Windows, macOS)
+        :return:
+            str - A str of detected operating system, e.g. "linux", "win32", "darwin"
+        """
+        return sys.platform
+
+    def find_all_pen_drives(self) -> Optional[list]:
+        """
+        Detects and lists all connected pen drives across different operating systems (Linux, Windows, macOS).
+        It determines the current operating system and delegates the task to an OS-specific function.
+        :return:
+            list - A list of detected pen drive mount points (partitions or whole drives).
+            None - if no pen drives are detected.
+        """
+        if self.check_os() == "linux":
+            return self.find_all_pen_drives_linux()
+        elif self.check_os() == "win32":
+            return self.find_all_pen_drives_win()
+        elif self.check_os() == "darwin":
+            pass  # TODO implement macos function
+
+    @staticmethod
+    def find_all_pen_drives_linux() -> Optional[list]:
```

Comment:
you can make find_all_pen_drives_linux private(protected) method as it should only be called from the class by public find_all_pen_drives method

same goes for find_all_pen_drives_win method

P.S. in Python if you start name of method or variable with underscore it means you want to make it protected for example _find_all_pen_drives_linux